File: android/app/src/main/java/com/liuhx/jswebview/injector/GetPackageNameHandler.java
package com.liuhx.jswebview.injector;

import android.webkit.WebView;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author liuhx
 * @date 2018/7/10.
 */

public class GetPackageNameHandler implements JsHandler {
    @Override
    public String action() {
        return "getPackageName";
    }

    @Override
    public void handleJs(WebView webView, JSONObject object) {
        JSONObject toJs = new JSONObject();
        try {
            String name = webView.getContext().getPackageName();
            toJs.put("result", "ok");
            JSONObject data = new JSONObject();
            data.put("name", name);
            toJs.put("data", data);
        } catch (JSONException e) {
            e.printStackTrace();
            return;
        }
        WebViewInjector.invokeCallback(webView, object, toJs);
    }
}

File: ios/JsWebView/JsFramework/GetPackageNameHandler.m
//
//  GetStorageInfoHandler.m
//  JsWebView
//
//  Created by liuhx on 2018/7/11.
//  Copyright © 2018 hursing. All rights reserved.
//

#import "GetPackageNameHandler.h"

@implementation GetPackageNameHandler

- (NSString *)action {
    return @"getPackageName";
}

- (void)handleJsFromWebView:(WKWebView *)webView info:(NSDictionary *)info {
    NSString *name = [[NSBundle mainBundle] bundleIdentifier];
    NSMutableDictionary *toJs = [[NSMutableDictionary alloc] initWithCapacity:4];
    toJs[@"result"] = @"ok";
    toJs[@"data"] = @{@"name": name};
    invokeCallback(webView, info, toJs);
}

@end
